### PR TITLE
refactor: integrated spotifyApi able to login

### PR DESCRIPTION
**Pull request by ImPatriciaLofamia**

### CODE REVIEW

**Review by bsistoso — COMMENTED** [review]
remove spaces

File: .env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+NEXTAUTH_URL=http://localhost:3000
+NEXT_PUBLIC_CLIENT_SECRET=081d6c6174ac4fe9925768f659335161
+NEXT_PUBLIC_CLIENT_ID=d7b5781f19594f3dbb588914750420ce
+JWT_SECRET=some_super_secret_value
```
Comment (bsistoso) [inline .env:4]: remove spaces

File: pages/api/auth/[...nextauth].tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  } catch (error) {
+    console.log(error)
+    return {
+      ...token,
+      error: "RefreshAccessTokenError",
+    }
+  }
+}
+export default NextAuth({
+  // Configure one or more authentication providers
```
Comment (bsistoso) [inline pages/api/auth/[...nextauth].tsx]: remove comments

File: pages/index.tsx [hunk 3]
```diff
@@ -1,11 +1,15 @@
 import React from "react";
 import SideBarNav from "@/components/SideBarNav";
+import Center from "@/components/Center";
 
 export default function Home() {
   return (
     <>
```
Comment (bsistoso) [inline pages/index.tsx]: remove unused tag

File: pages/Login.tsx [hunk 4]
```diff
@@ -1,24 +1,25 @@
 import React from "react";
+import { getProviders, signIn } from "next-auth/react";
 import Button from "@/components/Button";
 import InputField from "@/components/InputField";
 
-export const Login = () => {
+function Login({ providers }) {
```
Comment (bsistoso) [inline pages/Login.tsx]: make it arrow function

**Review by bsistoso — APPROVED** [review]

File: pages/Login.tsx [hunk 5]
```diff
@@ -101,50 +101,70 @@ export const Login = () => {
...
+              {Object.values(providers).map((provider: any) => {
+                return (
+                  <div key={provider.name}>
+                    <Button
+                      className="px-8 py-3"
+                      success
+                      buttonName="LOGIN"
+                      onClick={() => {
+                        signIn(provider.id, { callbackUrl: "/" });
+                        console.log("uadad", provider);
```
Comment (jarapamikaella) [inline pages/Login.tsx]: remove logs, use alert instead

File: pages/Login.tsx [hunk 6]
```diff
@@ -1,25 +1,25 @@
...
 
-export const Login = () => {
+export const Login = (providers) => {
   return (
-    <div className="flex w-full justify-center flex-col">
-      <div className="w-full flex justify-center border-b-2 border-gray-300 py-6">
+    <div className="flex flex-col justify-center w-full">
+      <div className="flex justify-center w-full py-6 border-b-2 border-gray-300">
         <div className="flex flex-row items-center">
           <svg viewBox="0 0 50 50" width="40px" height="40px">
```
Comment (jarapamikaella) [inline pages/Login.tsx]: move to an svg component

File: components/Button.tsx [hunk 7]
```diff
@@ -1,31 +1,38 @@
 import React from "react";
-
+import Svg from "./Svg";
```
Comment (mika-ella) [inline components/Button.tsx]: remove if unused

File: components/Button.tsx [hunk 8]
```diff
@@ -1,31 +1,38 @@
...
-  icon?: any;
-  className?: string;
-  onClick?: () => void;
-  success?: boolean;
-  primary?: boolean;
+  buttonName: string,
+  icon?: any,
+  className?: string,
+  onClick?: () => void,
+  src? : string,
```
Comment (mika-ella) [inline components/Button.tsx]: what's the use of this property? src?

File: components/Button.tsx [hunk 9]
```diff
@@ -1,31 +1,38 @@
...
+            onClick={onClick}
+            className={`${className} rounded-full flex justify-center py-1 border border-gray-400 sm:text-sm items-center font-medium
             ${success && "text-black bg-green-400  hover:bg-green-600"}
             ${primary && "text-white bg-blue-500 hover:bg-blue-600"}
             `}
         >
-            <span className="pr-1 hidden sm:block">
+            <span className="hidden sm:block">
                 {icon}
+                <img
```
Comment (mika-ella) [inline components/Button.tsx]: is this also for an icon?

File: components/Svg.tsx [hunk 10]
```diff
@@ -0,0 +1,13 @@
+interface SvgProps {
+    src : string,
+    className: string
+}
+const Svg = ({src, className} : SvgProps) => {
```
Comment (mika-ella) [inline components/Svg.tsx]: this is not an SVG component, image is used for this, if you are not going to use an SVG,
you can directly call an image element in the component that use this

File: pages/Login.tsx [hunk 11]
```diff
@@ -1,150 +1,113 @@
...
-          </svg>
+          <Svg
```
Comment (mika-ella) [inline pages/Login.tsx]: use image instead

File: pages/Login.tsx [hunk 12]
```diff
@@ -1,150 +1,113 @@
...
-                    fill="url(#Ld6sqrtcxMyckEl6xeDdMa)"
-                    d="M24,4C12.954,4,4,12.954,4,24s8.954,20,20,20s20-8.954,20-20S35.046,4,24,4z"
-                  />
-                  <path
-                    fill="#fff"
-                    d="M26.707,29.301h5.176l0.813-5.258h-5.989v-2.874c0-2.184,0.714-4.121,2.757-4.121h3.283V12.46 c-0.577-0.078-1.797-0.248-4.102-0.248c-4.814,0-7.636,2.542-7.636,8.334v3.498H16.06v5.258h4.948v14.452 C21.988,43.9,22.981,44,24,44c0.921,0,1.82-0.084,2.707-0.204V29.301z"
-                  />
-                </svg>
-              }
+              src="fb.png"
```
Comment (mika-ella) [inline pages/Login.tsx]: this is the same as the icon, instead use imageIcon property

File: pages/Login.tsx [hunk 13]
```diff
@@ -1,150 +1,113 @@
...
-                    text-anchor="none"
-                  >
-                    <g transform="scale(5.12,5.12)">
-                      <path d="M44.52734,34.75c-1.07812,2.39453 -1.59766,3.46484 -2.98437,5.57813c-1.94141,2.95313 -4.67969,6.64063 -8.0625,6.66406c-3.01172,0.02734 -3.78906,-1.96484 -7.87891,-1.92969c-4.08594,0.01953 -4.9375,1.96875 -7.95312,1.9375c-3.38672,-0.03125 -5.97656,-3.35156 -7.91797,-6.30078c-5.42969,-8.26953 -6.00391,-17.96484 -2.64844,-23.12109c2.375,-3.65625 6.12891,-5.80469 9.65625,-5.80469c3.59375,0 5.85156,1.97266 8.82031,1.97266c2.88281,0 4.63672,-1.97656 8.79297,-1.97656c3.14063,0 6.46094,1.71094 8.83594,4.66406c-7.76562,4.25781 -6.50391,15.34766 1.33984,18.31641zM31.19531,8.46875c1.51172,-1.94141 2.66016,-4.67969 2.24219,-7.46875c-2.46484,0.16797 -5.34766,1.74219 -7.03125,3.78125c-1.52734,1.85938 -2.79297,4.61719 -2.30078,7.28516c2.69141,0.08594 5.47656,-1.51953 7.08984,-3.59766z"></path>
-                    </g>
-                  </g>
-                </svg>
-              }
-              className="bg-black text-white hover:bg-slate-700 px-8 py-3"
+              src="apple.png"
```
Comment (mika-ella) [inline pages/Login.tsx]: use image property src is too vague and unreadable

File: pages/Login.tsx [hunk 14]
```diff
@@ -1,150 +1,113 @@
...
+                return (
+                  <div key={provider.name}>
+                    <Button
+                      className="px-8 py-3"
+                      success
+                      buttonName="LOGIN"
+                      onClick={() => {
+                        signIn(provider.id, { callbackUrl: "/" });
+                        alert(`DASURV KO NG FREE MILKTEA 🥰`);
+                        //console.log("uadad", provider);
```
Comment (mika-ella) [inline pages/Login.tsx]: remove

File: pages/Login.tsx [hunk 15]
```diff
@@ -1,150 +1,113 @@
...
+              {Object.values(providers).map((provider: any) => {
+                return (
+                  <div key={provider.name}>
+                    <Button
+                      className="px-8 py-3"
+                      success
+                      buttonName="LOGIN"
+                      onClick={() => {
+                        signIn(provider.id, { callbackUrl: "/" });
+                        alert(`DASURV KO NG FREE MILKTEA 🥰`);
```
Comment (mika-ella) [inline pages/Login.tsx]: modify alert message

Comment (mika-ella) [reply pages/Login.tsx]: or remove

File: tailwind.config.js [hunk 16]
```diff
@@ -4,12 +4,14 @@ module.exports = {
...
   ],
   theme: {
     extend: {},
   },
-  plugins: [],
-}
\ No newline at end of file
+  plugins: [
+    require('tailwind-scrollbar-hide'),
+  ]
```
Comment (mika-ella) [inline tailwind.config.js]: remove upper space

File: components/Button.tsx [hunk 17]
```diff
@@ -2,34 +2,44 @@ import React from "react";
...
-            <span className="pr-1 hidden sm:block">
-                {icon}
-            </span>
-            {buttonName}
-        </button>
-    )
+    >
+      <span>
+        {imageIcon}
+        <img src={image} className={`w-0 hidden sm:block ${image && "w-7"}`} />
```
Comment (mika-ella) [inline components/Button.tsx:38]: make it as optional also since it's optional in props

File: components/InputField.tsx [hunk 18]
```diff
@@ -1,12 +1,13 @@
...
-  type: string;
-  value?: string;
-  onChange?: (text: string) => void;
-  label?: string;
+  placeholder?: string,
+  className?: string,
+  type: string,
+  value?: any,
+  label?: string,
+  min? : any,
```
Comment (mika-ella) [inline components/InputField.tsx]: is this property used in this component?

File: components/InputField.tsx [hunk 19]
```diff
@@ -1,12 +1,13 @@
...
+  max? : any
```
Comment (mika-ella) [inline components/InputField.tsx]: is this property used in this component?

File: components/TrackPlayer.tsx [hunk 20]
```diff
@@ -0,0 +1,175 @@
...
+export const TrackPlayer = () => {
+  const spotifyApi = useSpotify();
+  const { data: session, status } = useSession();
+  const [currentTrackId, setCurrentTrackId] =
+    useRecoilState(currentTrackIdState);
+  const [isPlaying, setIsPlaying] = useRecoilState(isPlayingState);
+  const [volume, setVolume] = useState(50);
+  const songInfo = useSongInfo();
+
+  const fetchCurrentSong = () => {
```
Comment (mika-ella) [inline components/TrackPlayer.tsx:33]: move to hooks

File: components/TrackPlayer.tsx [hunk 21]
```diff
@@ -0,0 +1,175 @@
...
+      spotifyApi.getMyCurrentPlayingTrack().then((data) => {
+        setCurrentTrackId(data.body?.item?.id);
+
+        spotifyApi.getMyCurrentPlaybackState().then((data) => {
+          setIsPlaying(data.body?.is_playing);
+        });
+      });
+    }
+  };
+  const handlePlayPause = () => {
```
Comment (mika-ella) [inline components/TrackPlayer.tsx:44]: move to hooks

File: components/TrackPlayer.tsx [hunk 22]
```diff
@@ -0,0 +1,175 @@
...
+      debouncedAdjustVolume(volume);
+    }
+  }, [volume]);
+  const debouncedAdjustVolume = useCallback(
+    debounce((volume) => {
+      spotifyApi.setVolume(volume).catch((err) => {});
+    }, 500),
+    []
+  );
+  useEffect(() => {
```
Comment (mika-ella) [inline components/TrackPlayer.tsx:88]: duplication of useEffect, this one is already present at line 61-65

File: pages/home.tsx [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+
+  useEffect(() => {
+    if (!session?.user && status !== "loading" && status !== "authenticated") {
+      router.push("/login");
+    }
+  }, [session]);
+
+  useEffect(() => {
+    if (spotifyApi.getAccessToken()) {
+      spotifyApi.getUserPlaylists().then((data: any) => {
```
Comment (mika-ella) [inline pages/home.tsx:29]: move to hooks

File: pages/Login.tsx [hunk 24]
```diff
@@ -1,150 +1,109 @@
 import React from "react";
-import { getProviders, signIn } from "next-auth/react"
+import { getProviders, signIn } from "next-auth/react";
 import Button from "@/components/Button";
 import InputField from "@/components/InputField";
 
-export const Login = () => {
+function Login({providers}) {
```
Comment (mika-ella) [inline pages/Login.tsx:6]: convert to arrow function

File: lib/time.tsx [hunk 25]
```diff
@@ -0,0 +1,9 @@
+export function millisToMinutesAndSeconds(millis) {
+    const minutes = Math.floor(millis / 60000);
+    const seconds = ((millis % 60000) / 1000).toFixed(0);
+  
+    return (seconds == 60
```
Comment (mika-ella) [inline lib/time.tsx:5]: move the returned value to a variable, do not perform the condition in the return statement

File: components/TrackPlayer.tsx [hunk 26]
```diff
@@ -0,0 +1,175 @@
...
+                onClick={() => volume < 100 && setVolume(volume + 10)}
+              />
+            }
+          />
+          <Button
+            buttonName=""
+            imageIcon={
+              <MusicNoteIcon
+                className="w-5 h-5 cursor-pointer hover:scale-125 transition transform duration-100 ease-out hidden md:block"
+                onClick={() => {}}
```
Comment (mika-ella) [inline components/TrackPlayer.tsx:182]: is onClick a required prop? if not, remove it.

File: components/SideBarNav.tsx [hunk 27]
```diff
@@ -1,54 +1,82 @@
...
 
 const SideBarNav = () => {
+  const spotifyApi = useSpotify();
+  const { data: session, status } = useSession();
+  const [playlists, setPlaylists] = useState([]);
+  const [playlistId, setPlaylistId] = useRecoilState(playlistIdState);
+
+  useEffect(() => {
+    if (spotifyApi.getAccessToken()) {
+      spotifyApi.getUserPlaylists().then((data) => {
```
Comment (mika-ella) [inline components/SideBarNav.tsx:25]: move to hooks

**Review by jarapamikaella — APPROVED** [review]

**Review by mika-ella — APPROVED** [review]

**Pull request merged**